entrypoint.sh

```diff
@@ -22,18 +22,39 @@ MESSAGE="${MESSAGE:-}"
 NAME=$(jq ".pusher.name" "$GITHUB_EVENT_PATH")
 EMAIL=$(jq ".pusher.email" "$GITHUB_EVENT_PATH")
 
+[[ "$GITHUB_EVENT_NAME" == "delete" ]] && event_ref=$(jq --raw-output .ref "$GITHUB_EVENT_PATH") || event_ref=""
+
+BUILD_ENV_VARS="${BUILD_ENV_VARS:-}"
+
+ENV_VARS=""
+JSON_BUILD_ENV_VARS=""
+for BUILD_ENV_VAR in $BUILD_ENV_VARS; do
+    BUILD_ENV_KEY=$(echo "$BUILD_ENV_VAR" | cut -d '=' -f1 | awk '{ print toupper($0) }')
+    BUILD_ENV_VALUE=$(echo "$BUILD_ENV_VAR" | cut -d '=' -f2)
+    ENV_VARS+=" --arg $BUILD_ENV_KEY $BUILD_ENV_VALUE" 
+    JSON_BUILD_ENV_VARS+="\"$BUILD_ENV_KEY\": \$$BUILD_ENV_KEY,"$'\n'
+done
+    
 # Use jq’s --arg properly escapes string values for us
 JSON=$(
   jq -n \
     --arg COMMIT  "$COMMIT" \
     --arg BRANCH  "$BRANCH" \
     --arg MESSAGE "$MESSAGE" \
+    --arg GITHUB_EVENT_NAME "$GITHUB_EVENT_NAME" \
+    --arg EVENT_REF "$event_ref" \
     --arg NAME    "$NAME" \
     --arg EMAIL   "$EMAIL" \
+    $ENV_VARS \
     '{
       "commit": $COMMIT,
       "branch": $BRANCH,
       "message": $MESSAGE,
+      "env": {
+        "GITHUB_EVENT_NAME": $GITHUB_EVENT_NAME,
+        "GITHUB_EVENT_REF": $EVENT_REF,
+        '"$JSON_BUILD_ENV_VARS"'
```

Comment:
I'll have a go at doing something nicer. Just wanted to throw the basic idea up in case you guys were already thinking of it. 

Thanks for the feedback!

Reply:
Nice! Yeah I think you can smoosh two JSON objects together with JQ:

```bash
$ jq -s '.[0] * .[1]' <(echo '{"a":1}') <(echo '{"b":2}')
{
  "a": 1,
  "b": 2
}
```

Something like:

```bash
#!/bin/bash

GITHUB_EVENT_NAME="push"
GITHUB_EVENT_REF="refs/tag/v1.3.0"
BUILD_ENV_VARS='{"CUSTOM_ENV": "value"}'

GITHUB_ENV_JSON=$(
  jq -n \
    --arg GITHUB_EVENT_NAME "$GITHUB_EVENT_NAME" \
    --arg GITHUB_EVENT_REF "$GITHUB_EVENT_REF" \
    '{
      GITHUB_EVENT_NAME: $GITHUB_EVENT_NAME,
      GITHUB_EVENT_REF: $GITHUB_EVENT_REF,
    }'
)

BUILD_ENV_JSON=$(
  jq -s '.[0] * .[1]' \
    <(echo "$GITHUB_ENV_JSON") \
    <(echo "$BUILD_ENV_VARS")
)

echo "$BUILD_ENV_JSON"
# =>
# {
#   "GITHUB_EVENT_NAME": "push",
#   "GITHUB_EVENT_REF": "refs/tag/v1.3.0",
#   "CUSTOM_ENV": "value"
# }
```

Reply:
Ah awesome, and jq will validate the passed in `BUILD_ENV_VARS` are valid for free? Nice!

Will get that change made soon

Reply:
Looking ace!